src/features/session_trace/aggregate/index.js

```diff
@@ -1,538 +1,195 @@
-
-    resources.forEach((currentResource) => {
-      if ((currentResource.fetchStart | 0) <= this.#laststart) return // don't recollect already-seen resources
-
-      const parsed = parseUrl(currentResource.name)
-      const res = {
-        n: currentResource.initiatorType,
-        s: currentResource.fetchStart | 0,
-        e: currentResource.responseEnd | 0,
-        o: parsed.protocol + '://' + parsed.hostname + ':' + parsed.port + parsed.pathname, // resource.name is actually a URL so it's the source
-        t: currentResource.entryType
-      }
-      this.storeSTN(res)
-    })
-
-    this.#laststart = resources[resources.length - 1].fetchStart | 0
-  }
-
-  // JavascriptError (FEATURE) events pipes into ST here.
-  storeErrorAgg (type, name, params, metrics) {
-    if (type !== 'err') return // internal errors are purposefully ignored
-    const node = {
-      n: 'error',
-      s: metrics.time,
-      e: metrics.time,
-      o: params.message,
-      t: params.stackHash
-    }
-    this.storeSTN(node)
-  }
-
-  // Ajax (FEATURE) events--XML & fetches--pipes into ST here.
-  storeXhrAgg (type, name, params, metrics) {
-    if (type !== 'xhr') return
-    const node = {
-      n: 'Ajax',
-      s: metrics.time,
-      e: metrics.time + metrics.duration,
-      o: params.status + ' ' + params.method + ': ' + params.host + params.pathname,
-      t: 'ajax'
+    /** Only start actually harvesting if running in full mode at init time */
+    if (this.mode === MODE.FULL) this.startHarvesting()
+    else {
+      /** A separate handler for noticing errors, and switching to "full" mode if running in "error" mode */
+      registerHandler('stn-errorAgg', () => {
+        if (this.mode === MODE.ERROR) this.switchToFull()
+      }, this.featureName, this.ee)
     }
-    this.storeSTN(node)
+    this.agentRuntime.session.write({ sessionTraceMode: this.mode })
+    this.drain()
   }
 
-  // Central function called by all the other store__ & addToTrace API to append a trace node.
-  storeSTN (stn) {
-    if (this.nodeCount >= this.maxNodesPerHarvest) { // limit the amount of pending data awaiting next harvest
-      if (this.isStandalone || this.agentRuntime.session.state.sessionTraceMode !== MODE.ERROR) return
-      const openedSpace = this.trimSTNs(ERROR_MODE_SECONDS_WINDOW) // but maybe we could make some space by discarding irrelevant nodes if we're in sessioned Error mode
-      if (openedSpace === 0) return
-    }
-
-    if (this.isStandalone && now() >= MAX_TRACE_DURATION) {
-      return
-    }
-
-    if (this.trace[stn.n]) this.trace[stn.n].push(stn)
-    else this.trace[stn.n] = [stn]
-
-    this.nodeCount++
+  /** This module does not auto harvest by default -- it needs to be kicked off.  Once this method is called, it will then harvest on an interval */
+  startHarvesting () {
+    if (this.scheduler.started || this.scheduler.aborted || this.blocked) return
+    this.scheduler.runHarvest({ needResponse: true })
```

Comment:
{needResponse: true} is not needed anymore; can take out here & take it out of the trace.js

Reply:
`this.scheduler.aborted` is also extraneous since that's only set by stoptime(true)